glommio/src/io/dma_file.rs

```diff
@@ -406,15 +406,13 @@ impl DmaFile {
         self.file.path()
     }
 
-    /// Convenience method that closes a DmaFile wrapped inside an Rc
-    pub async fn close_rc(self: Rc<DmaFile>) -> Result<()> {
+    /// Convenience method that closes a DmaFile wrapped inside an Rc.
+    ///
+    /// Returns whether the file is actually closed, or just dropped an Rc.
+    pub async fn close_rc(self: Rc<DmaFile>) -> Result<bool> {
```

Comment:
It would be clearer to introduce an enum, something like

```rs
#[derive(Debug)]
enum CloseResult {
    Closed,
    Unreferenced,
    Error(Err),
}
```

This could be reused in `ImmutableFile::close`.

Reply:
This looks weird to me. If we want to give the user that information we should do so through a `Result` so that we can propagate normally using `?`. i.e.

```
#[derive(Debug)]
enum CloseFailedError {
    Unreferenced,
    CloseFailed(GlommioError),
}
pub async fn close_rc(self: Rc<DmaFile>) -> Result<CloseResult> { ... }
```

Reply:
One of those cases isn't an error, so shouldn't be treated as such. Not being able to use `?` is an inconvenience, but eventually we'll be able to implement [`Try`](https://github.com/rust-lang/rust/issues/84277).

Reply:
That's a good point. An enum is more clear than a boolean.